scalafix/rules/src/main/scala/fix/Cats_v1_0_0.scala

```diff
@@ -29,160 +14,105 @@ case class RemoveCartesianBuilder(index: SemanticdbIndex)
     Symbol("_root_.cats.syntax.CartesianOps.`|@|`.") ::
       (1 to 22).toList.map(arity =>
       Symbol(
-        s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.`|@|`.")): _*)
+        s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.`|@|`.")): _*
+  )
 
   private[this] val partialApplies = SymbolMatcher.normalized(
     Symbol(s"_root_.cats.syntax.CartesianOps.`*>`."),
     Symbol(s"_root_.cats.syntax.CartesianOps.`<*`.")
   )
 
-  private[this] val renames: Map[Symbol, String] =
-    (1 to 22)
-      .map { arity =>
-        Seq(
-          Symbol(
-            s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.map.") -> "mapN",
-          Symbol(
-            s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.imap.") -> "imapN",
-          Symbol(
-            s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.contramap.") -> "contramapN"
-        )
-      }
-      .flatten
-      .toMap
+  private[this] val renames: Map[String, String] = (1 to 22).flatMap { arity =>
+    Seq(
+      s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.map." -> "mapN",
+      s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.imap." -> "imapN",
+      s"_root_.cats.syntax.CartesianBuilder.CartesianBuilder$arity.contramap." -> "contramapN"
+    )
+  }.toMap
 
   private[this] val cartesianOps =
-    SymbolMatcher.normalized(renames.keys.toSeq: _*)
+    SymbolMatcher.normalized(renames.keys.map(Symbol.apply).toSeq: _*)
 
-  // Hackish way to work around duplicate fixes due to recursion
-  val alreadyFixedOps = collection.mutable.Set.empty[Term.Name]
   private[this] def replaceOpWithComma(ctx: RuleCtx, op: Term.Name): Patch =
-    if (op.matches(cartesianBuilders) && !alreadyFixedOps.contains(op)) {
-      alreadyFixedOps += op
+    // replace |@| with ,
+    ctx.replaceTree(op, ",") ++
       // remove the space before |@|
-      ctx.removeToken(ctx.tokenList.prev(op.tokens.head)) +
-        // replace |@| with ,
-        ctx.replaceTree(op, ",")
-    } else {
-      Patch.empty
-    }
-
-  private[this] def removeCartesianBuilderOp(
-      ctx: RuleCtx,
-      applyInfix: Term.ApplyInfix): Patch = {
-    applyInfix match {
-      case Term.ApplyInfix(lhs: Term.ApplyInfix, op, _, _) =>
-        removeCartesianBuilderOp(ctx, lhs) + replaceOpWithComma(ctx, op)
-      case Term.ApplyInfix(_, op, _, _) =>
-        replaceOpWithComma(ctx, op)
-    }
-  }
+      ctx.tokenList
+        .leading(op.tokens.head)
+        .takeWhile(_.is[Whitespace])
+        .map(ctx.removeToken)
 
   private[this] def wrapInParensIfNeeded(ctx: RuleCtx, t: Term): Patch = {
-    if (t.tokens.head.is[Token.LeftParen] && t.tokens.last
-        .is[Token.RightParen]) {
-      Patch.empty
-    } else {
-      ctx.addLeft(t.tokens.head, "(") + ctx.addRight(t.tokens.last, ")")
-    }
-  }
+    for {
+      head <- t.tokens.headOption
```

Comment:
are there concrete cases in which `head` is not safe?

Reply:
Yes, it's quite common unfortunately :/ 

```scala
@ "trait A".parse[Source].get.collect {
    case t @ Name(_) => t.tokens.head.syntax
  }
java.util.NoSuchElementException: next on empty iterator
```

Reply:
wut, why isn't `A` the head of those tokens?

Reply:
There's a `Name.Anonymous()` inside the Template.self
```scala
@ template"{}"
res3: Template = Template(List(), List(), Self(_, None), List())
```